### PR TITLE
Set b:undo_ftplugin

### LINKED ISSUE #21: b:undo_ftplugin not set in filetype plugin
All filetype plugins should set `b:undo_ftplugin` to a command string that, when evaluated upon a filetype change, undoes any configuration changes made by the plugin.

See `:help undo_ftplugin` for details and [examples](https://github.com/vim/vim/search?q=b%3Aundo_ftplugin) in the distribution.

### PR DESCRIPTION
**Pull request by dkearns**
Filetype plugins should set b:undo_ftplugin to a command string that, when evaluated upon a filetype change, will undo any configuration changes made by the plugin.  See :help undo_ftplugin for details.

Fixes #21.

@krischik, just ignore this for the moment if you like.  It needs testing and further work.

### CODE REVIEW

**Comment by krischik** [comment]
Looking a the draft: I could not have done that in this quality. Looking forward to the finished pull request.

**Comment by krischik** [comment]
I just noted to late: pull requests into the develop branch would have been better. That way I can test the pull request before it goes out to the world.

File: ftplugin/ada.vim [hunk 1]
```diff
@@ -98,6 +110,7 @@ if !exists ("b:match_words")  &&
       \ '\%(\<while\>.*\|\<for\>.*\|'.s:notend.'\)\<loop\>:\<end\s\+loop\>,' .
       \ '\%(\<do\>\|\<begin\>\):\<exception\>:\<end\%(\s*\%($\|;\)\|\s\+\%(\%(if\|case\|loop\|record\)\>\)\@!\a\)\@=,' .
       \ s:notend . '\<record\>:\<end\s\+record\>'
+   let b:undo_ftplugin .= " | unlet! b:match_words"
    let b:match_skip = 's:Comment\|String\|Operator'
```
Comment (krischik) [inline ftplugin/ada.vim:114]: If I correctly understand we concatenate command into `b:undo_ftplugin` that remove all buffer variables again. With the other, already merged, pull request a new variable `b:match_skip` was added. Should that variable be undone as well?

**Review by krischik — APPROVED** [review]
I have done the other two merge request first as they where not in draft status. Then I addressed the merge conflicts. I hope I got everything correct. I have one question but it's not a blocker.

Thank you for the work done.

File: ftplugin/ada.vim (hunk 1 above)
Comment (dkearns) [reply ftplugin/ada.vim:114]: Yes, it should.  I'll update this.

**Comment by krischik** [comment]
Thank you!

**Pull request merged**